File: 506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        sorted_scores = sorted(enumerate(score), key=lambda x: x[1], reverse=True)

        result = [""] * len(score)
        
        for rank, (original_index, _) in enumerate(sorted_scores):
            if rank == 0:
                result[original_index] = "Gold Medal"
            elif rank == 1:
                result[original_index] = "Silver Medal"
            elif rank == 2:
                result[original_index] = "Bronze Medal"
            else:
                result[original_index] = str(rank + 1)

        return result

File: 506-relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int n = score.length;
        int[][] arr = new int[n][2];
        for(int i=0;i<n;i++){
            arr[i][0] = score[i];
            arr[i][1] = i;
        }
        Arrays.sort(arr, (a, b) -> b[0] - a[0]);
        String[] ans = new String[n];
        for(int i=0;i<n;i++){
            int temp = arr[i][1];
            if(i == 0) ans[temp] = "Gold Medal";
            else if(i == 1) ans[temp] = "Silver Medal";
            else if(i == 2) ans[temp] = "Bronze Medal";
            else ans[temp] = String.valueOf(i + 1);
        }
        return ans;
    }
}